zhusuan/model/base.py

```diff
@@ -283,15 +283,18 @@ def _check_names_exist(self, name_or_names):
 
         :param name_or_names: A string or a list of strings. Names of
             `StochasticTensor` s in the network.
+        :return: The validated name, or a tuple of the validated names.
         """
-        if isinstance(name_or_names, (tuple, list)):
-            names = name_or_names
+        if isinstance(name_or_names, six.string_types):
+            names = (name_or_names,)
```

Comment:
will return tuple if a single string is passed in.

Reply:
@korepwx

Reply:
@thjashin _check_names_exists will return `name_or_names`, not `names`.

Reply:
https://stackoverflow.com/questions/68630/are-tuples-more-efficient-than-lists-in-python